utility/retrieve_polarion_ids.py

```diff
@@ -0,0 +1,141 @@
+usage = """Retrieve Polarion Ids from test suites.
+
+
+In the context of CephCI,
+    path     full path to suites directory
+
+  Example:
+    python retrieve_polarion_ids.py <path> --exclude luminous,nautilus
+
+Usage:
+  retrieve_polarion_ids.py <path> [--exclude <str>]
+  retrieve_polarion_ids.py -h | --help
+
+Options:
+  --exclude <pattern>   patterns to be excluded
+  -h --help             Show this screen
+
+"""
+import logging
+import sys
+from os import walk
+from os.path import join
+
+from docopt import docopt
+from yaml import dump, safe_load
+
+
+def is_excluded(regex, string):
+    """Return true if one of the excluded pattern matches.
+
+    Args:
+        regex: patterns for comparison
+        string
+    Returns:
+        boolean True if matches else False
+    """
+    for exc in regex:
+        if exc in string:
+            return True
+    return False
+
+
+def retrieve_polarion_data(file_string):
+    """Retrieve polarion data
+
+    Return Test cases
+    - with multiple Ids along with test case name, file
+    - with no Ids along with test case name, file
+
+          - test:
+              name: Install ceph pre-requisites
+              desc: installation of ceph pre-requisites
+              module: install_prereq.py
+              abort-on-fail: true
+
+    Args:
+        file_string: file path
+    """
+
+    multiple_ids = []
+    missing_ids = []
+    good = []
+    _total = 0
+
+    with open(file_string, "r") as fs:
+        ts = safe_load(fs)
+        for tc in ts["tests"]:
+            _tc = tc["test"]
+            tc_name = _tc["name"]
+            if "install_prereq.py" in _tc["module"]:
+                continue
+            _total += 1
+
+            if not _tc.get("polarion-id"):
+                missing_ids.append(tc_name)
+                continue
+            pol_ids = _tc["polarion-id"]
+            pol_ids = pol_ids.split(",")
+
+            if len(pol_ids) > 1:
+                multiple_ids.append({tc_name: pol_ids})
+            else:
+                good.append({tc_name: pol_ids[-1]})
+
+    return {
+        "multiple_ids": multiple_ids,
+        "missing_ids": missing_ids,
+        "good": good,
+        "total": _total,
+    }
+
+
+if __name__ == "__main__":
+    logging.basicConfig(
```

Comment:
Do we need `logging` as we are mainly using `print` statements? We could move all print statements to `log`

Reply:
done